packages/core/src/html/NodeProcessor.js

```diff
@@ -88,17 +89,70 @@ class NodeProcessor {
     return node.type === 'text' || node.type === 'comment';
   }
 
+  static getVslotShorthand(node) {
+    if (!node.attribs) {
+      return undefined;
+    }
+    const keys = Object.keys(node.attribs);
+    return _.find(keys, key => key.startsWith('#'));
+  }
+
   /**
    * Processes the markdown attribute of the provided element, inserting the corresponding <slot> child
-   * if there is no pre-existing slot child with the name of the attribute present.
+   * if there is no pre-existing slot child with the name of the attribute present. For existing <slot>
+   * children, the HTML node tag name will be converted to "template".
    * @param node Element to process
    * @param attribute Attribute name to process
    * @param isInline Whether to process the attribute with only inline markdown-it rules
    * @param slotName Name attribute of the <slot> element to insert, which defaults to the attribute name
    */
   _processAttributeWithoutOverride(node, attribute, isInline, slotName = attribute) {
     const hasAttributeSlot = node.children
-      && node.children.some(child => _.has(child.attribs, 'slot') && child.attribs.slot === slotName);
+      && node.children.some((child) => {
+        /*
+          User can use latest Vue slot shorthand syntax (#test) or deprecated Vue slot syntax (slot="test").
+          If user has both, we will prioritize processing the shorthand syntax.
+        */
+        const vslotShorthand = NodeProcessor.getVslotShorthand(child);
+        const hasSlot = _.has(child.attribs, 'slot');
+
+        if (vslotShorthand) {
+          const vslotShorthandName = vslotShorthand.substring(1, vslotShorthand.length); // remove #
+          const isTargetVslotShorthand = vslotShorthandName === slotName;
+          if (hasSlot) {
+            // User has both kinds of syntax
+            const isTargetSlot = child.attribs.slot === slotName;
+            if (isTargetSlot && isTargetVslotShorthand) {
+              /*
+                Both syntax's slot names match what we are looking for.
+                We will use the the shorthand syntax and delete the deprecated syntax.
+              */
+              delete child.attribs.slot;
+            }
+          }
+          if (isTargetVslotShorthand) {
+            /*
+              User may use tag names like "div" or "p" but we have to change it to "template"
+              for vslot shorthand syntax to work, according to Vue2.6 syntax documenatation.
```

Comment:
you'll need to "shift it one level deeper" in the dom tree instead (then wrap with `<template #xx>`) as the `<p slot="..."` syntax dosen't erase the `p` element.